### PR TITLE
DOCS 5677 - changing all references from otel to opentelemetry

### PR DESCRIPTION
**Pull request by cswatt**
somebody should definitely review this and make sure i haven't changed anything that wasn't supposed to be changed (an "otel" in a link, parameter name, etc.)

but otherwise ready to merge

### CODE REVIEW

**Comment by hestonhoffman** [comment]
I made a [ticket](https://datadoghq.atlassian.net/browse/DOCS-5709) for this one, cos my brains are fried.

File: content/en/opentelemetry/guide/_index.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ further_reading:
   text: "Datadog's partnership with OpenTelemetry"
 - link: "https://www.datadoghq.com/blog/monitor-otel-with-w3c-trace-context/"
   tag: "Blog"
-  text: "Monitor OTel-instrumented apps with support for W3C Trace Context"
+  text: "Monitor-instrumented apps with support for W3C Trace Context"
```
Comment (kayayarai) [inline content/en/opentelemetry/guide/_index.md:12]:
```suggestion
  text: "Monitor OpenTelemetry-instrumented apps with support for W3C Trace Context"
```

File: content/en/real_user_monitoring/connect_rum_and_traces/_index.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ further_reading:
     text: "Troubleshoot with Session Replay browser dev tools"
   - link: "https://www.datadoghq.com/blog/correlate-traces-datadog-rum-otel/"
     tag: "Blog"
-    text: "Correlate Datadog RUM events with traces from OTel-instrumented applications"
+    text: "Correlate Datadog RUM events with traces from-instrumented applications"
```
Comment (kayayarai) [inline content/en/real_user_monitoring/connect_rum_and_traces/_index.md:25]:
```suggestion
    text: "Correlate Datadog RUM events with traces from OpenTelemetry-instrumented applications"
```

**Review by kayayarai — COMMENTED** [review]
Not finished review, just have to save

**Review by kayayarai — APPROVED** [review]
Made a couple fixes, approving.

**Pull request merged**